[Ryan Warwick Han] iP

RyanWarwick opened the pull request:
## Chappy

> “Manners. Maketh. Man.” – Harry Hart ([source](https://www.youtube.com/watch?v=xT0M2YxFQ50))

Let this handsome chap do your scheduling for you. It would be his honour.

- Gets things done in a jiffy
- Makes your schedule look spiffy
- Will never get you in a tiffy

You can make yourself Chappy's priority just by

1. download the jar file from [here](https://github.com/RyanWarwick/ip/releases/tag/A-Release).
2. double click it.
3. tell him your tasks.
4. let Chappy handle it with distinguished poise :tophat:

Best of all, he will work for ~~FREE~~ some chips and beer!

Features:

- [x] Managing tasks
- [x] Managing deadlines, events
- [ ] Reminders (coming soon)
- [x] Searching tasks and free dates
- [x] Load and save tasks
- [x] Manly bada** gentleman

Here's a simple code snippet to `print` an ASCII tophat for your troubles

```
public class Tophat {
    public static void main(String[] args) {
        String hat = """
                                :..
                               #@@@@#+:
                              -@@@@@@@@@*-
                             .@@@@@@@@@@@@@#-
                             %@@@@@@@@@@@@@@@%=
                      -==:. *#@@@@@@@@@@@@@@@@@@.
                     .@@@@@%=--*@@@@@@@@%%%%@@%-
                      .#@@@#-::::=#@@@@%%%@@%-
                        :#@@@#-::...:+#%@@%=
                          .+%@@%*-     :*=
                             :+%@@@*=-=%%=
                                :=#@@@@@@@%.
                                    :=*#@@@*
                                    """;
        System.out.println(hat);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

RyanWarwick committed:
Edit README.md
Edit docs/README.md
Remove Duke.java
Add Chappy.java
Replace Duke references with Chappy

RyanWarwick committed:
Chappy.java: Implement java.util.Scanner to read user input
Chappy.java: Echo user input if not "bye"
Chappy.java: Add while loop to await user input
Chappy.java: Break while loop and exit on "bye" user input

RyanWarwick committed:
Chappy.java: Add static String array to store user input
Chappy.java: Print String array contents (if not null) on "list" input

RyanWarwick committed:
Chappy.java: Change userInputArray to use Task object instead of String
Chappy.java: Add mark/unmark feature for Tasks
Task.java: Create Task class

RyanWarwick committed:
Chappy.java: Replace Task objects with Todo, Event and Deadline
Todo.java: Create Todo class
Deadline.java: Creat Deadline class
Event.java: Create Event class

RyanWarwick committed:
Chappy.java: Fix typo in print formatting
input.txt: Add test command
EXPECTED.TXT: Add expected output from running test command

RyanWarwick committed:
Chappy.java: Add error handling for missing description or information
when adding tasks

RyanWarwick committed:
Chappy.java: Add delete feature
Chappy.java: Change userInputArray from Task array to ArrayList
Chappy.java: Fix bug allowing for empty description and information (leading/trailing whitespace)

RyanWarwick committed:
Chappy.java: Fix Java Scanner object throwing NoSuchElementException
Chappy.java: Fix Java Scanner object not reading next input.txt line(s)

fusin02 commented on src/main/java/Chappy.java:
```diff
@@ -0,0 +1,156 @@
+         :+%@@@*=-=%%=        
+            :=#@@@@@@@%.      
+                :=*#@@@*      
+ .----------------.  .----------------.  .----------------.  .----------------.  .----------------.  .----------------.
```
This visual looks amazing, makes the user experience much better.

fusin02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good job with the naming, makes things very clear

fusin02 reviewed (commented):
Good job with the coding standards bro, looks great

fusin02 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    private String completeBy;
```
Perhaps you can name this "by"? Might make it easier for you

fusin02 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    private String from;
```
Great naming, makes it super clear.

RyanWarwick committed:
CreateTaskException.java: Implement custom exception for create task
failure
Task.java: Shift create and remove task actions to static methods
Deadline.java: Implement Enums for command input options
Deadline.java: Shift command input option validation to Class
Event.java: Implement Enums for command input options
Event.java: Shift command input option validation to Class
Chappy.java: Implement Enums for command options
Chappy.java: Change command detection to switch statement
Chappy.java: Handle user input errors using custom exceptions

RyanWarwick committed:
Implement saving of task list to disk in json format
Implement loading of task list from disk on app run

RyanWarwick committed:
Merge branch 'branch-Level-7'

slidings commented on src/main/java/Chappy.java:
```diff
@@ -0,0 +1,244 @@
+            return;
+        }
+
+        String logo = """
```
this looks pretty cool!! 🔥

slidings commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,26 @@
+        return j;
+    }
+
+    public static Task fromJson(JSONObject jsonObject) {
```
All of the naming of variables and methods are really clear and intuitive. Good job 👍

slidings commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,75 @@
+import org.json.simple.JSONObject;
+
```
I also only just realised that there isn't javadocs in most classes/methods. (Which demonstrates how good your naming is)

slidings commented on src/main/java/Chappy.java:
```diff
@@ -0,0 +1,244 @@
+        }
+    }
+
+    public static void saveToDisk() throws IOException {
```
Maybe in 20 years those kids won't know whats a disk  /s🤣 
Perhaps can consider local storage if you like?

slidings reviewed (commented):
Really clean formatting and adherence to the naming convention 🔥 
Could add some javadoc on complex classes/methods to make it easier to understand for other reviewers or your future self.
Otherwise good job!

RyanWarwick committed:
Deadline.java: Add parsing of user input to LocalDate object
Event.java: Add parsing of user input to LocalDate object
Deadline.java: Move option validation to factory method
Event.java: Move option validation to factory method
Chappy.java: Move option validation to Class factories

RyanWarwick committed:
Merge branch 'branch-level-8'

RyanWarwick committed:
Ui.java: Add Ui Class to handle user inputs
Parser.java: Add Parser Class to handle types of Commands
Storage.java: Add Storage Class to handle loading and saving of data
TaskList.java: Add TaskList Class to handle CRUD operations on data

RyanWarwick committed:
Merge branch 'branch-more-oop'

RyanWarwick committed:
Move classes to respective chappy.* packages

RyanWarwick committed:
Merge branch 'branch-packages'

RyanWarwick committed:
Merge remote-tracking branch 'origin/add-gradle-support'

RyanWarwick committed:
Storage.java: Fix check for empty JSON object when parsing from disk
EventTest.java: Add unit test for of() Event factory method
StorageTest.java Add unit test for loadFromDisk() method
build.gradle: Add dependency for json-simple v1.1.1
Remove json-simple jar file from project

RyanWarwick committed:
build.gradle: Fix application mainClass value

RyanWarwick committed:
Add JavaDoc comments for most public methods

RyanWarwick committed:
Chappy.java: Remove indentation for switch-case
Remove trailing whitespaces in files

RyanWarwick committed:
TaskList.java: Add findTask method to list Task(s) containing matching
keyword in values
Chappy.java: Add FIND switch-case for find command keyword search feature.
Parser.java: Add FIND Command enum type for find command

RyanWarwick committed:
Merge branch 'branch-A-JavaDoc'

RyanWarwick committed:
Merge branch 'branch-A-CodingStandard'

RyanWarwick committed:
Merge branch 'branch-Level-9'

RyanWarwick committed:
Fix bug when saving Task done status to disk

RyanWarwick committed:
Change user interaction to GUI-based

RyanWarwick committed:
Merge branch 'branch-Level-10'

RyanWarwick committed:
Add assert statements for important assumptions in code

RyanWarwick committed:
Merge pull request #2 from RyanWarwick/branch-A-Assertions

Add assert statements for important assumptions in code

RyanWarwick committed:
Merge branch 'master' into branch-A-CodeQuality

RyanWarwick committed:
Refactor code to improve code quality

RyanWarwick committed:
Merge pull request #3 from RyanWarwick/branch-A-CodeQuality

Branch a code quality

RyanWarwick committed:
Add feature to find nearest x free days

The Free command allows the user to find the nearest day where the next
x days are not allocated to Tasks. e.g. If there is an Event from
23-09-2024 to 25-10-2024, and the user wants to find the nearest 4 free
days from today (26-09-2024), then the result will be 26-10-2024.

This command currently only works with Days because there is only
implementation of LocalDate and not LocalDateTime. Let's work towards
making the Free command compatible with other options such as Hours.

RyanWarwick committed:
Merge pull request #4 from RyanWarwick/branch-BCD-Extension

Add feature to find nearest x free days

RyanWarwick committed:
Add User Guide

The User Guide is extremely for introducing the application to new
users, especially the relevant commands and examples of commands and use
cases.

RyanWarwick committed:
Fix issue with Ui.PNG in README.md

RyanWarwick committed:
Fix naming syntax for Ui.png

RyanWarwick committed:
Set file extension for Ui.png to lowercase
